crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -273,6 +273,15 @@ pub unsafe trait SystemParam: Sized {
         world: UnsafeWorldCell<'world>,
         change_tick: Tick,
     ) -> Self::Item<'world, 'state>;
+
+    /// Returns true if this [`SystemParam`] requires exclusive access to the
+    /// entire [`World`]. [`System`]s with exclusive parameters cannot run in
+    /// parallel with other systems.
+    ///
+    /// [`System`]: crate::system::System
+    fn is_exclusive() -> bool {
```

Comment:
Out of curiosity, what is the need for this compared with just setting the Access to `mut_all_components`? (which would mean that the system is incompatible with any other system)?

Is it for optimization reasons that we want to 'pre-compute' this?

Reply:
The only way to know if a system is exclusive is based on its parameters now, since we don't have two separate implementations on `System` now.